src/osd/ECUtil.cc

```diff
@@ -71,28 +68,55 @@ int ECUtil::decode(
     need.insert(i->first);
   }
 
-  for (uint64_t i = 0; i < total_data_size; i += sinfo.get_chunk_size()) {
+  set<int> avail;
+  for (auto i = to_decode.begin();
+       i != to_decode.end();
+       ++i) {
+    assert(i->second.length() != 0);
+    avail.insert(i->first);
+  }
+
+  map<int, vector<pair<int, int>>> min;
+  assert(ec_impl->minimum_to_decode(need, avail, &min) == 0);
+
+  int chunks_count = 0;
+  int repair_data_per_chunk;
+  int subchunk_size = sinfo.get_chunk_size()/ec_impl->get_sub_chunk_count();
+
+  for (auto &&i : to_decode) {
+    if (min.find(i.first) == min.end()) continue;
```

Comment:
nit, this is an anti-pattern: 
1. find and throw away the returned iterator, 
2. use `operator[]` to locate the element to find again.

also, `j` deserves a better name. could you change it to something like:
```c++
auto found = min.find(i.first);
if (found != min.end()) {
  int repair_subchunk_count = 0;
  for (auto& subchunks : found) {
    repair_subchunk_count += subchunks.second;
  }
  break;
}
```